### DISCUSSION TITLE
Support for WinUI 3 Desktop

### DISCUSSION THREAD

**Discussion by MarkBQE** [PrismLibrary/Prism#2450] [General]
@brianlagunas are there any plans to add support for WinUI 3 desktop?

**Comment by dansiegel** [comment]
WinUI 3 is already supported as part of Prism 8.1 thanks to the work by @jeromelaban

**Reply by MarkBQE** [reply]
This is great news for Prism and the community. When are we planning to release Prism 8.1?  @jeromelaban thanks for all your hard work.

**Reply by MarkBQE** [reply]
@dansiegel  @brianlagunas. Quick question  to get access to Prism 8.1, I should be a sponsor. I have no issues with that, who should I sponsor Dan or Brian? Please confirm we have Prism implementation for WINUI3.0 desktop handled in 8.1. Thanks in advance.

**Reply by brianlagunas** [reply]
Sponsor whomever you want. Doesn't matter.

**Reply by MarkBQE** [reply]
@jeromelaban, @dansiegel  Please confirm we have handled WinUI3.0 Desktop  Prism implementation in Prism 8.1

**Reply by dansiegel** [reply]
@MarkBQE you can see in the [e2e app](https://github.com/PrismLibrary/Prism/tree/master/e2e/Uno/WinUI) all of the platforms that @jeromelaban has tested which includes Windows desktop

**Reply by MarkBQE** [reply]
Appreciate the update.

**Reply by MarkBQE** [reply]
@jeromelaban FYI after upgrading to Project Reunion .8, Region Manager no longer works. Throws an exception

'Method not found: 'Microsoft.System.DispatcherQueue Microsoft.UI.Xaml.DependencyObject.get_DispatcherQueue()'.'.

I have a content control defined as a region in my view. This was working fine with Project Reunion 0.5


<ContentControl  VerticalAlignment="Stretch" HorizontalAlignment="Stretch" VerticalContentAlignment="Stretch" HorizontalContentAlignment="Stretch"
                    prismRegions:RegionManager.RegionName="{Binding RegionName}" />

**Reply by jeromelaban** [reply]
Thanks. On which platform are you seeing this error? Note that Reunion 0.8 added some subtle breaking changes that we have not yet added support for (in Uno specifically), this should come soon.

**Reply by MarkBQE** [reply]
@jeromelaban  I am getting this error with WinUI 3 desktop specifically using the package "Prism.DryIoc.Uno.WinUI" 

<PackageReference Include="Microsoft.ProjectReunion" Version="0.8.0" />
    <PackageReference Include="Microsoft.ProjectReunion.Foundation" Version="0.8.0" />
    <PackageReference Include="Microsoft.ProjectReunion.WinUI" Version="0.8.0" />
    <PackageReference Include="Prism.DryIoc.Uno.WinUI" Version="8.1.97" />

Content control is defined as a region in the shell like so, I believe we need to update some references

<UserControl
    x:Class="WinUI3Prism8.Views.Shell"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:WinUI3Prism8.Views"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:prism="using:Prism"
                xmlns:prismMvvm="using:Prism.Mvvm"
                xmlns:prismRegions="using:Prism.Regions"
    mc:Ignorable="d">

    <Grid x:Name="ShellGrid" VerticalAlignment="Stretch" HorizontalAlignment="Stretch">
        <ContentControl  VerticalAlignment="Stretch" HorizontalAlignment="Stretch" VerticalContentAlignment="Stretch" HorizontalContentAlignment="Stretch"
              prismRegions:RegionManager.RegionName="{Binding RegionName}"    />

        
    </Grid>
</UserControl>

**Reply by MarkBQE** [reply]
> Thanks. On which platform are you seeing this error? Note that Reunion 0.8 added some subtle breaking changes that we have not yet added support for (in Uno specifically), this should come soon.

@jeromelaban any progress on adding support to the breaking changes that creeped in Reunion 0.8

**Reply by jeromelaban** [reply]
We've been looking at the Reunion 0.8 update, and as far as Uno is concerned, nothing needed to change. The latest builds from Prism also have been [upgraded to 0.8](https://github.com/PrismLibrary/Prism/blob/281535d32ceebdac3b84d8c95218a7255da8630d/Packages.props#L38). The error you're seeing is likely not related to reunion.

If you have a repro app that shows the issue, that will help diagnose the issue.

**Reply by MarkBQE** [reply]
@jeromelaban   I have created a  simple app for your reference. I have a region defined in the shell view, when the region is commented the app works fine. However as soon as I build and run after I uncomment the region the app crashes. I hope this simple app helps you get to the bottom of this. BTW the same code works fine with 0.5, this is an issue only with Reunion 0.8

https://github.com/MarkBQE/TestWINUIForJerome

Thanks in advance for looking into this.

**Reply by jeromelaban** [reply]
This was addressed by https://github.com/PrismLibrary/Prism/pull/2530 for Uno will follow as well when updating to WinAppSDK 1.0, and is adjusted when updating to a later build of Prism.

**Reply by MarkBQE** [reply]
Thanks for looking into this over the weekend, appreciate your commitment. I will wait for Prism V8.2 to be released.

**Reply by MarkBQE** [reply]
@dansiegel  when is Prism 8.2 scheduled to be released, I would like to test @jeromelaban's fix related to DispatcherQueue namespace issue introduced in Project Union 1.0

**Reply by dansiegel** [reply]
As has been discussed for virtually every release, we do not ship Prism based on timelines. When we feel we have enough to warrant a release then we do a release. If you need real time updates then I would suggest that you help keep Prism sustainable by stepping up as a Sponsor and gaining access to the private feed on [Sponsor Connect](https://sponsorconnect.dev) as well as access to the Sponsor Only Discord Channel.
